libs/designsystem/src/lib/components/checkbox/checkbox.component.spec.ts

```diff
@@ -1,105 +1,125 @@
   });
 
   it('should create', () => {
     expect(spectator.component).toBeTruthy();
   });
 
-  describe('event: checkedChange', () => {
-    it('should emit true', () => {
-      spyOn(spectator.component.checkedChange, 'emit');
-      const change = true;
+  describe('when configured', () => {
+    describe('by default', () => {
+      it('should have attention level 2', () => {
+        expect(spectator.component.attentionLevel).toBe('2');
+        expect(spectator.element).toHaveClass('attention-level2');
+      });
 
-      spectator.component.onChecked(change);
+      it('should not be checked', () => {
+        expect(spectator.component.checked).toBe(false);
+      });
 
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledTimes(1);
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledWith(true);
+      it('should not be disabled', () => {
+        expect(spectator.component.disabled).toBe(false);
+      });
+
+      it('should not have error', () => {
+        expect(spectator.component.hasError).toBe(false);
+      });
     });
 
-    it('should emit false', () => {
-      spyOn(spectator.component.checkedChange, 'emit');
-      const change = false;
+    it('should have class .attention-level1 when attentionLevel is 1', () => {
+      const attentionLevel = '1';
 
-      spectator.component.onChecked(change);
+      spectator.setInput('attentionLevel', attentionLevel);
+      spectator.detectChanges();
 
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledTimes(1);
-      expect(spectator.component.checkedChange.emit).toHaveBeenCalledWith(false);
+      expect(spectator.element).toHaveClass('attention-level1');
     });
-  });
 
-  describe('Inputs', () => {
-    it('should bind the class .square to ion-checkbox', () => {
-      const squareShape = 'square';
-      spectator.component.ngOnChanges({
-        shape: new SimpleChange(null, squareShape, false),
-      });
+    it('should set the [checked] input on ion-checkbox to true', () => {
+      const checked = true;
 
-      spectator.setInput('shape', squareShape);
+      spectator.setInput('checked', checked);
 
-      expect(spectator.query('ion-checkbox')).toHaveClass('square');
-      expect(spectator.query('ion-checkbox')).not.toHaveClass('circle');
+      expect(checkbox.checked).toBe(true);
     });
 
-    it('should bind the class .circle to ion-checkbox', () => {
-      const circleShape = 'circle';
-      spectator.component.ngOnChanges({
-        shape: new SimpleChange(null, circleShape, false),
-      });
+    it('should set the [checked] input on ion-checkbox to false', () => {
+      const checked = false;
 
-      spectator.setInput('shape', circleShape);
+      spectator.setInput('checked', checked);
 
-      expect(spectator.query('ion-checkbox')).toHaveClass('circle');
-      expect(spectator.query('ion-checkbox')).not.toHaveClass('square');
+      expect(checkbox.checked).toBe(false);
     });
```

Comment:
Does this actually test anything? Isn't the `checked` state `false` by default? Consider setting to true and assert, then set to false and assert...